### PR TITLE
Added new command: sort_task

### LINKED ISSUE #152: New command: general sort

### LINKED ISSUE #158: Update DG: Storage

### PR DESCRIPTION
**Pull request by SoulUseless**
Added relevant unit tests (TaskComparatorTest), and augmented current unit tests for classes that I added new functionality to.
Updated relevant section of UG

closes #152; - will open another one for sort_event, and sort_task unit testing for Sochedule and ModelManager

### CODE REVIEW

**Comment by SoulUseless** [comment]
Added DG updates to the PR:
- Created skeleton DG for future edits
- Updated AB3 Storage documentation to Sochedule Storage documentation (closes #158)
- Added Implementation details of sortTask into DG

File: src/main/java/seedu/address/logic/parser/SortTaskCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.logic.commands.SortTaskCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+public class SortTaskCommandParser implements Parser<SortTaskCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the DoneTaskCommand
```
Comment (ddanelynn) [inline src/main/java/seedu/address/logic/parser/SortTaskCommandParser.java]: Javadoc not matching the method!! You may wanna update

**Review by ddanelynn — APPROVED** [review]
LGTM except a minor javadoc mismatch!!

**Review by ljhgabe — APPROVED** [review]
LGTM

**Pull request merged**